cfme/tests/middleware/test_middleware_deployment.py

```diff
@@ -12,21 +12,50 @@
 pytest_generate_tests = testgen.generate(testgen.provider_by_type, ["hawkular"], scope="function")
 
 
+@pytest.mark.usefixtures('has_no_middleware_providers')
 def test_list_deployments(provider):
-    """Tests deployments count between UI, DB and Management system
+    """Tests deployments list between UI, DB and Management system
+    This test requires that no any other provider should exist before.
 
     Steps:
+        * Creates a provider.
         * Get deployments list from UI
         * Get deployments list from Database
         * Get deployments list from Management system(Hawkular)
         * Compare size of all the list [UI, Database, Management system]
     """
-    ui_deps = MiddlewareDeployment.deployments(provider=provider)
-    db_deps = MiddlewareDeployment.deployments_in_db()
-    mgmt_deps = MiddlewareDeployment.deployments_in_mgmt(provider)
-    assert len(ui_deps) == len(db_deps) == len(mgmt_deps), \
-        ("Size of deployments mismatch! UI:{}, DB:{}, MGMT:{}"
-         .format(len(ui_deps), len(db_deps), len(mgmt_deps)))
+    provider.create(cancel=False, validate_credentials=False)
```

Comment:
@hhovsepy As I mentioned in an earlier commit, we need to put a comment as to why we are doing it this way. The "better" way to do it is as I said, to make sure that the calls are provider specific, but if that is not possible right now, let's add a comment as to why we are doing the provider create locally. I think @mfalesni was just thinking that as a "provider" is a prerequisite to a test, that we should probably have it as a fixture, and not inside the actual test. So maybe have a fixture that creates the provider that uses the has_no_middleware_providers fixture as its prerequisite.